File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/VerifierApi.swift
import Foundation

/// Facade for online credential verification operations.
public struct VerifierApi {
    let verifier: Verifier

    public func verifyCredential(_ credential: VerifiableCredential) async throws -> Bool {
        try await verifier.verifyCredential(credential: credential)
    }

    public func verifySignature(did: String, keyId: String, signature: Data, data: Data) async throws -> Bool {
        try await verifier.verifySignature(did: did, keyId: keyId, signature: signature, data: data)
    }

    public func resolveDid(_ did: String) async throws -> DidDocument {
        try await verifier.resolveDid(did: did)
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/VerifierApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.DidDocument
import my.ssdid.sdk.domain.model.VerifiableCredential
import my.ssdid.sdk.domain.verifier.Verifier

class VerifierApi internal constructor(private val verifier: Verifier) {
    suspend fun verifyCredential(credential: VerifiableCredential): Result<Boolean> =
        verifier.verifyCredential(credential)
    suspend fun verifySignature(did: String, keyId: String, signature: ByteArray, data: ByteArray): Result<Boolean> =
        verifier.verifySignature(did, keyId, signature, data)
    suspend fun resolveDid(did: String): Result<DidDocument> = verifier.resolveDid(did)
}
